src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/configuration/RabbitMQConfiguration.kt

```diff
@@ -45,4 +47,49 @@ data class ConnectionManagerConfiguration(
         check(workingThreads > 0) { "expected 'workingThreads' greater than 0 but was $workingThreads" }
         check(!confirmationTimeout.run { isNegative || isZero }) { "expected 'confirmationTimeout' greater than 0 but was $confirmationTimeout" }
     }
+
+    fun createRetryingDelaySequence(): Sequence<RetryingDelay> {
+        return generateSequence(RetryingDelay(0, minConnectionRecoveryTimeout)) {
+            RetryingDelay(it.tryNumber + 1, RetryingDelay.getRecoveryDelay(
+                it.tryNumber + 1,
+                minConnectionRecoveryTimeout,
+                maxConnectionRecoveryTimeout,
+                maxRecoveryAttempts,
+                retryTimeDeviationPercent
+            ))
+        }
+    }
+
+}
+
+data class RetryingDelay(val tryNumber: Int, val delay: Int) {
+    companion object {
+        fun getRecoveryDelay(
```

Comment:
Please, add `@JvmStatic` annotation for method in order to correctly use in from Java